src/main/java/org/jenkinsci/plugins/gitclient/RemoteGitImpl.java

```diff
@@ -196,17 +197,17 @@ public void clearCredentials() {
 
     /** {@inheritDoc} */
     public void addCredentials(String url, StandardCredentials credentials) {
-        proxy.addCredentials(url, credentials); // credentials are Serializable
+        proxy.addCredentials(url, CredentialsProvider.snapshot(StandardCredentials.class, credentials)); // credentials are Serializable
```

Comment:
The supplied `clazz` is just used to maintain the type signature.

```
        Class bestType = null;
        CredentialsSnapshotTaker bestTaker = null;
        for (CredentialsSnapshotTaker taker : ExtensionList.lookup(CredentialsSnapshotTaker.class)) {
            if (clazz.isAssignableFrom(taker.type()) && taker.type().isInstance(credential)) {
                if (bestTaker == null || bestType.isAssignableFrom(taker.type())) {
                    bestTaker = taker;
                    bestType = taker.type();
                }
            }
        }
        if (bestTaker == null) {
            return credential;
        }
        return clazz.cast(bestTaker.snapshot(credential));
```

So the best fit (with the highest ordinal in case of equivalent fits) will win.